Sofa/Component/Topology/Mapping/src/sofa/component/topology/mapping/SubsetTopologicalMapping.cpp

```diff
@@ -941,6 +941,106 @@ void SubsetTopologicalMapping::updateTopologicalMappingTopDown()
             break;
         }
 
+        case core::topology::TETRAHEDRAADDED:
+        {
+        }
+
+        case core::topology::TETRAHEDRAREMOVED:
+        {
+            if (!d_handleTetrahedra.getValue()) break;
+            if (!toTetrahedronMod) toModel->getContext()->get(toTetrahedronMod);
+            if (!toTetrahedronMod) break;
+
+            // tRem is a pointer to the tetra elements that should be removed
+            // this change comes from the topology container in the source topology.
+            // So here, we have to update our data that maps between source and destination
+            // tD2S and tS2D, and also update the topology container of this model (the destination)
+            const TetrahedraRemoved *tRem = static_cast< const TetrahedraRemoved * >( topoChange );
+
+            // each tetra is an integer that specifies it's index in the source topology
+            sofa::type::vector<Index> tetra_array_buffer_source = tRem->getArray();
+
+            // we want to remove the elements in both source and destination
+            sofa::type::vector<Index> tetra_array_buffer_destination;
+            tetra_array_buffer_destination.reserve(tetra_array_buffer_source.size());
+
+            // for each tetra that should be removed, look up the correct tetra in the destination topology
+            for (auto &tetra_source : tetra_array_buffer_source)
+            {
+                Index tetra_destination = tS2D[tetra_source];
+                if (tetra_destination == sofa::InvalidID)
+                    continue;
+                tetra_array_buffer_destination.push_back(tetra_destination);
+            }
```

Comment:
It feels like you are iterating twice over this array although you could have done it only once latter by appending the `tetra_array_buffer_destination` in a scoped if line 997

Reply:
Sure, I can change that. However, right now the changes are not applied correctly.
Do you see anything obvious that is missing? See error messages above.

Reply:
I don't know if that is the reason, but you are working with the datas `tS2D` ans `tD2S` but those are for triangles. You should be modifying the datas `teS2D` and `teD2S`

Reply:
Ohhhhh, I see. That makes much more sense. :D 
Thanks! :)

Reply:
That solved some of the issues, but uncovered a much larger issue underneath. :D
See https://github.com/sofa-framework/sofa/issues/5127

Reply:
Ok, thank you about that, we will take a look, this seems quite serious indeed.